### PR TITLE
OLMIS-7657: Add table to mobile Stock on Hand page

### PR DESCRIPTION
**Pull request by dariapionk**
Added table to mobile stock on hand page

I added scss file and I was thinking about previous files added by me which had style defined in react components. Should I add their style in scss in this particular PR?

OLMIS-7657

[OLMIS-7657]: https://openlmis.atlassian.net/browse/OLMIS-7657?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/stock-on-hand-mobile/components/stock-on-hand-table-component.jsx [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+                            return (
+                                <tr className={`row-expanded ${expandedProducts.indexOf(row.original.orderable.id) == -1 && 'hidden'}`}>
+                                    <td className='cell-expanded'>
+                                        <div>Lot Code</div>
+                                        <div>Quantity</div>
+                                    </td>
+                                    <td/>
+                                    <td className='cell-expanded'>
+                                        <div>
+                                            {product.lot ? product.lot.lotCode : 'No lot defined'}
```
Comment (sniedzielski) [inline src/stock-on-hand-mobile/components/stock-on-hand-table-component.jsx:69]: I think this line could be written in such way using optional chaining:
                                   `         {product?.lot?.lotCode ??  'No lot defined'}`

**Review by sniedzielski — COMMENTED** [review]
Good job. I left  some suggestions, Please look at them. @dariapionk

**Comment by sniedzielski** [comment]
And regarding your question @dariapionk  - "Should I add their style in scss in this particular PR" yes if only you have a time. It could unified the way how styles are definied.

File: src/stock-on-hand-mobile/pages/stock-on-hand.jsx [hunk 2]
```diff
@@ -38,16 +48,88 @@ const StockOnHand = ({ facilityService, programService }) => {
...
+        });
+    };
+
+    const getExpandedProducts = () => {
+        return JSON.parse(localStorage.getItem('expandedProducts')) || [];
+    };
+
+    const handleExpandView = (productId) => {
+        const expandedProducts = getExpandedProducts();
+        const isExpanded = expandedProducts.indexOf(productId) != -1;
```
[the remark this answers is not in the document]
Comment (dariapionk) [reply src/stock-on-hand-mobile/pages/stock-on-hand.jsx:69]: @sniedzielski  -1 is returned by indexOf function when there is no such element in an array. I can change this expression to something like
`const isExpanded = expandedProducts.filter((product) => product.id == productId).length > 0`

[the remark this answers is not in the document]
Comment (sniedzielski) [reply src/stock-on-hand-mobile/pages/stock-on-hand.jsx:69]: Perfect, such statement could be more readable.

**Comment by dariapionk** [comment]
@sniedzielski I addressed your changes requests, could you take a look at it? I also added styles for previous compontents in scss file

**Review by sniedzielski — APPROVED** [review]
LGTM

**Pull request merged**